Fix bare metal builds on Cypress targets

LDong-Arm opened the pull request:
<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html).

NOTE: Do not remove any of the template headings (even for optional sections) as this
template is automatically parsed. 
-->

### Summary of changes <!-- Required -->

<!-- 
    Please provide the following information: 

    Description of the the change (what is this fixing / adding / removing?).

    Why the change is needed (if this is fixing a reported issue please summarize what
    the issue is and add the reference. E.g. Fixes #17119).

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
    
-->

Fixes: #13108

This PR fixes bare metal build of two Cypress targets:
* CYSBSYSKIT_01: `COMPONENT_SCL` depends on the EMAC/LWIP stack which is absent on bare metal. Add an `mbed_lib.json` to disable its build in the bare metal profile.
* CYTFM_064B0S2_4343W: Make `cytfm_flash*` dependent on TF-M and PSA presence.

#### Impact of changes <!-- Optional -->
<!-- 
    If there are any implications for users taking this change then they must be 
    provided here. For Major PR types this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

https://github.com/ARMmbed/mbed-os-example-blinky-baremetal now compiles successfully on those Cypress targets.

#### Migration actions required <!-- Optional -->
<!-- 
    This should only be applicable in Major PR types for which this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

None.

### Documentation <!-- Required -->

<!-- 
    Please provide details of any document updates required, including links to any
    related PRs against the docs repository.
    If no document updates are required please specify 'None', this at least tells us
    that this has been considered.
-->

None.

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

<!--
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front of them would change the meaning incorrectly. 
-->
    [x] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

<!--
    Provide all the information required, listing all the testing performed. For new targets please attach full test results for all supported compilers.
-->
    [] No Tests required for this change (E.g docs only update)
    [x] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->

<!--
    Request additional reviewers with @username or @team
-->

@MarceloSalazar @ARMmbed/mbed-os-core @ARMmbed/team-cypress

----------------------------------------------------------------------------------------------------------------

LDong-Arm committed:
CYTFM_064B0S2_4343W: use cytfm_flash only if TFM and PSA are present

This fixes bare metal builds

LDong-Arm commented on connectivity/drivers/emac/TARGET_Cypress/mbed_lib.json:
```diff
@@ -1,3 +1,8 @@
+    "name": "cypress-emac",
+    "target_overrides": {
+        "CYSBSYSKIT_01": {
+            "target.components_add": ["SCL"]
```
Alternatively, combine [targets/TARGET_Cypress/TARGET_PSOC6/COMPONENT_SCL](https://github.com/ARMmbed/mbed-os/tree/master/targets/TARGET_Cypress/TARGET_PSOC6/COMPONENT_SCL) into [connectivity/drivers/emac/TARGET_Cypress/COMPONENT_SCL](https://github.com/ARMmbed/mbed-os/tree/master/connectivity/drivers/emac/TARGET_Cypress/COMPONENT_SCL) so the code isn't scattered in two places?

kyle-cypress commented on targets/targets.json:
```diff
@@ -6765,7 +6765,6 @@
             "MCU_PSOC6_M4"
         ],
         "components_add": [
-            "SCL",
```
What I believe is a similar problem was encountered a while ago in TARGET_Cypress/TARGET_PSOC6/COMPONENT_WHD. The solution in that case was to add a trivial mbed_lib.json file under COMPONENT_WHD so that it is excluded in the bare metal build process by virtue of a.)being considered a library and b.)not being included in the bare metal profile's whitelist of libraries.

Would a similar approach be feasible here? I don't think this is necessarily wrong, but it would be better to be consistent if we can.

LDong-Arm replied:
That sounds better, let me try it.

LDong-Arm replied:
Done

kyle-cypress replied:
Looks good, thanks

LDong-Arm committed:
Add mbed_lib.json for PSOC6 SCL to exclude it from build metal build

SCL is for the Cypress Ethernet driver which is not currently
compatible with bare metal.
